connectivity/suite.go

```diff
@@ -17,47 +19,64 @@ type Hooks interface {
 	AddConnectivityTests(ct *check.ConnectivityTest) error
 }
 
-func Run(ctx context.Context, ct *check.ConnectivityTest, extra Hooks) error {
-	if err := ct.SetupAndValidate(ctx, extra); err != nil {
+func Run(ctx context.Context, connTests []*check.ConnectivityTest, extra Hooks) error {
+	if err := setupConnectivityTests(ctx, connTests, extra); err != nil {
 		return err
 	}
 
-	ct.Infof("Cilium version: %v", ct.CiliumVersion)
+	connTests[0].Infof("Cilium version: %v", connTests[0].CiliumVersion)
 
-	if ct.Params().Perf {
-		if err := builder.NetworkPerformanceTests(ct); err != nil {
+	builders, err := builder.GetTestSuites(connTests[0].Params())
```

Comment:
This is where `builder` could instead be `suiteBuilder` to make the overall logic much more readable.

e.g.

```
suiteBuilders, err := builder.GetTestSuites(connTests[0].Params())
```

Reply:
Renamed.